multi_import/importer.py

```diff
@@ -396,21 +434,57 @@ def process_row(self, row, data, context):
             errors = get_errors(ex)
             row.set_errors(errors)
 
-        data.processed = True
+    def process_row_subsequent_pass(self, row, data, context, serializer_class):
+        if row.errors:
+            return
 
-    def validate_row_post_save(self, row, data):
-        if not data.processed:
+        if not data.instance:
+            return
+
+        serializer = serializer_class(
+            instance=data.instance,
+            data=row.data.copy(),
+            context=context,
+            partial=data.instance is not None,
+        )
+
+        does_not_have_changes = not serializers.might_have_changes(serializer)
+
+        if does_not_have_changes:
+            data.add_diff(serializers.get_diff_data(serializer, no_changes=True))
+            return
+
+        is_valid = serializer.is_valid()
+
+        if not is_valid:
+            row.set_errors(serializer.errors)
+            return
+
+        if not serializers.has_changes(serializer):
```

Comment:
`not serializers.has_changes(...)` here and `does_not_have_changes` above =.=.. potentially quite misleading.. maybe just add a comment before the one above.